app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettingsFragment.kt

```diff
@@ -37,6 +37,8 @@ class WallpaperSettingsFragment : Fragment() {
         requireComponents.useCases.wallpaperUseCases
     }
 
+    private var snackBar: FenixSnackbar? = null
```

Comment:
This looks wrong to me.

> Apparently, `snackbars` usually exist alongside with coordinator layout, that would disable them automatically. Switching to Compose stripped away the benifit, hid the `snackbar` manually upon hearing `onStop()`.

If what you say is true, why not use the fragment view instead of using the `ComposeView` in the [`onWallpaperSelected`](https://github.com/mozilla-mobile/fenix/blob/189b22fb69922d7e52dc89bb82c791e795c2fef5/app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettingsFragment.kt#L98)? This view reference is used only for the snackbar.

Reply:
I followed the approach but had to adjust it. Fragment view in our case was a `ComposeView` (because `onCreateView()` was returning it). Changing in to the Coordinator did the trick!

Thanks for bringing it up!

Reply:
now that I think of it, I should add a comment on why the coordinator layout was used.

Reply:
We could also shift the snackbar creation back to compose. For example, we could do something like this:

```
fun WallpaperSettings(
    wallpaperGroups: Map<Wallpaper.Collection, List<Wallpaper>>,
    defaultWallpaper: Wallpaper,
    loadWallpaperResource: suspend (Wallpaper) -> Bitmap?,
    selectedWallpaper: Wallpaper,
    onSelectWallpaper: (Wallpaper) -> Unit,
    onLearnMoreClick: (String, String) -> Unit,
) {
    ...
                WallpaperThumbnails(
                    wallpapers = wallpapers,
                    defaultWallpaper = defaultWallpaper,
                    loadWallpaperResource = loadWallpaperResource,
                    selectedWallpaper = selectedWallpaper,
                    onSelectWallpaper = {
                        val result = onSelectWallpaper()
                        showWallpaperSnackbar(result)
                    },
                )

                Spacer(modifier = Modifier.height(32.dp))
            }
        }
    }
}
```

this would keep the snackbar attached to the compose tree and hopefully avoid the problem. It would also help to contain the settings view logic to a single file.

Reply:
Good point. But it would bring back the queuing bug we used to have (also the ui and animation is a bit inconsistent with the FenixSnackbar, but it's not a blocker for me). I would actually consider shifting back to compose, but may be not in the scope of this fix.

here is the differences in the UI between compose we had and FenixSnackbar (and the queuing issue)

[compose.webm](https://user-images.githubusercontent.com/92760693/193153074-2d0b4cf3-c4df-459c-ab02-1a7630f07ce0.webm)
[not-compose-2.webm](https://user-images.githubusercontent.com/92760693/193153040-aab17275-7690-406a-ae1d-127c5a4cd4b7.webm)

Reply:
Okay, in favor of getting a small patch through that fixes the crasher I am cool with landing this. Would you mind filing an issue to track shifting this snackbar back to Compose? It could potentially be a follow-up to https://github.com/mozilla-mobile/fenix/issues/23221.

It turns out the queue bug is pretty simple to deal with. We just need to dismiss the previous one when we dispatch a new one, something like:

```
onSelectWallpaper = { updatedWallpaper ->
    val messageText = when(onSelectWallpaper(updatedWallpaper)) {
        Wallpaper.ImageFileState.Downloaded -> ...
        Wallpaper.ImageFileState.Error -> ...
    }
    coroutineScope.launch {
        scaffoldState.snackbarHostState.currentSnackbarData?.dismiss()
        scaffoldState.snackbarHostState.showSnackbar(
            message = messageText,
            duration = SnackbarDuration.Short,
        )
    }
},
```